Automattic/wp-calypso
# SMP: add expire notice on grid tile Thumbnails

#### Proposed Changes

On #68965 A message was added for the Sites Management Page on list mode.
This PR introduces the same concept for Grid mode, a.k.a. Tiles.

* Add new component `SitesGridAction` to display messages in grid mode.
* Add new component `SitesGridActionRenew` to show the Renew CTA on expiring sites.
  * I'm using the same events used on `sites-plan-renew-nag.tsx`, I just added a new property `display_mode`.
  * I'm using the same icon on `sites-plan-renew-nag.tsx`.
  * I've added the plan name to the message.

#### Screenshots

1. `SitesGridActionRenew` on Coming Soon Site





<img width="439" alt="Screenshot 2022-10-25 at 15 06 19" src="https://user-images.githubusercontent.com/779993/197798616-d4e20d39-575c-4594-b313-49631d58766a.png">





2. `SitesGridActionRenew` on Public Site






<img width="452" alt="Screenshot 2022-10-25 at 15 07 10" src="https://user-images.githubusercontent.com/779993/197798127-ccfe637e-fee1-44a3-8c7d-a20eb18be617.png">


3. `SitesGridActionRenew` on Private Site






<img width="440" alt="Screenshot 2022-10-25 at 15 06 11" src="https://user-images.githubusercontent.com/779993/197798045-604d1aba-8885-40b5-bb64-fabcab85f8e6.png">



4. `SitesGridAction` with long text:








<img width="455" alt="Screenshot 2022-10-25 at 15 15 03" src="https://user-images.githubusercontent.com/779993/197797903-c23e268e-cf3c-453e-b31a-4df735fa6d2a.png">


#### Testing Instructions

* Select an Atomic site and change the expiration date to the past using SA.
* Go to `/sites`
* Search for your site
* Observe a Notice appears over the Thumbnail.
* Only if you are the site owner you'll see a `Renew` CTA. Other users will see only the notice without the link.
* Repeat these steps for coming soon, private and public sites.

#### Pre-merge Checklist



- Closes https://github.com/Automattic/dotcom-forge/issues/1080
- Related to "#68965" which introduced the same concept for list mode
